presentation/frontend/src/test/StreamedResponseGenerator.test.ts

```diff
@@ -0,0 +1,183 @@
+import { StreamedResponseGenerator, StreamedResponseGeneratorProps } from "../presentation-frontend/StreamedResponseGenerator";
+import { expect } from "chai";
+import sinon from "sinon";
+import { PageOptions } from "@itwin/presentation-common";
+
+async function sleep(millis: number) {
+  return new Promise<void>((resolve) => setTimeout(resolve, millis));
+}
+
+describe("StreamedResponseGenerator", () => {
+  it("should run requests concurrently", async () => {
+    const items = [...new Array(1000).keys()];
+    const props: StreamedResponseGeneratorProps<number> = {
+      getBatch: async (page) => {
+        await sleep(5);
+        return {
+          total: items.length,
+          items: items.slice(page.start, page.start + 10),
+        };
+      },
+    };
+
+    const generator = new StreamedResponseGenerator(props);
+    await generator.getItems();
+  }).timeout(100);
```

Comment:
This may fail sporadically, needs to be rewritten in a way that doesn't rely on timeouts. Suggestion:

1. make `getBatch` a `sinon.fake` that returns `ResolvablePromise` of a batch.
2. call `generator.getItems()`, don't await.
3. confirm the fake created in step 1 has been called expected number of times.
4. resolve all resolvable promises.
5. await on `getItems()` result, verify it.

Reply:
Make sense, done